std/functional.d

```diff
@@ -899,6 +903,89 @@ template partial(alias fun, alias arg)
     assert(result2 == 1.5f);
 }
 
+/**
+Takes a function of (potentially) many arguments, and returns a function taking
+one argument and returns a callable taking the rest.  f(x, y) == curry(f)(x)(y)
+
+Params:
+    F = a function taking at least one argument
+Returns:
+    A single parameter callable object
+*/
+template curry(alias F)
+{
+	//adapted from the implementation from Artur Skawina here:
+	//https://forum.dlang.org/post/mailman.1626.1340110492.24740.digitalmars-d@puremagic.com
+	auto curry()
+	{
+		struct CurryImpl {
+			import std.traits;
```

Comment:
Imports inside of structs should be selective to make the reader aware of the locally used symbols.

Reply:
Fixed